classes/OpeningHours/Module/Shortcode/IsOpen.php

```diff
@@ -81,12 +83,45 @@ public function shortcode ( array $attributes ) {
 
     $attributes['is_open'] = $isOpen;
     $attributes['classes'] .= ($isOpen) ? $attributes['open_class'] : $attributes['closed_class'];
-    $attributes['text'] = ($isOpen) ? $attributes['open_text'] : $attributes['closed_text'];
+
+    // If the attribute show_holiday is enabled
+    $closedText = $attributes['closed_text'];
+    if ($attributes['show_closed_holidays']) {
+        $holidaysList = $this->getTodaysHolidaysCommaSeperated($todayData);
+
+        $closedText = sprintf(
+            $holidaysList ? $attributes['closed_holiday_text'] : $closedText,
+            $holidaysList
+        );
+    }
+
+    $attributes['text'] = ($isOpen) ? $attributes['open_text'] : $closedText;
 
     echo $this->renderShortcodeTemplate($attributes, 'shortcode/is-open.php');
   }
 
   /**
+   * Retrieves holiday names for today
+   * @param  array $todayData   Data for today
+   * @return string            Extracted holiday name(s)
+   */
+  public function getTodaysHolidaysCommaSeperated($todayData) {
+      if (count($todayData['holidays']) > 0) {
+          $holidayNames = array();
+
+        foreach ($todayData['holidays'] as $holiday) {
+            array_push($holidayNames, $holiday->getName());
+        }
+
+        $names = implode(', ', $holidayNames);
+
+        return trim(rtrim($names, ','));
```

Comment:
I think this is redundant as implode only inserts commas between the entries or does this have any other purpose?

Reply:
Since we want to display the current holiday(s) where while it may be rare, there could be many, we comma separate them for display. I can instead change this implode out to instead append a string with the holiday name and a comma if you feel that works better?

```php

  public function getTodaysHolidaysCommaSeperated($todayData) {
      if (count($todayData['holidays']) > 0) {
          $holidayNames = '';;
        foreach ($todayData['holidays'] as $holiday) {
            $holidayNames .= $holiday->getName() . ', ';
        }
        return trim(rtrim($names, ','));
    }
    return false;
  }
```

Reply:
See my comment in the new change request. Didn't see that I can also reply here